mapwms.c

```diff
@@ -1139,7 +1145,55 @@ int msWMSLoadGetMapParams(mapObj *map, int nVersion,
       return msWMSException(map, nVersion, "InvalidFormat", wms_exception_format);
     }
   }
-  if (bboxfound && nVersion >= OWS_1_3_0) {
+  if (tiled > 0) {
+    const char *value;
+    double buffer;
+    hashTableObj *meta = &(map->web.metadata);
+
+    if ((value = msLookupHashTable(meta, "tile_map_edge_buffer")) != NULL) {
+        map_edge_buffer = atoi(value);
+    }
+    if (map_edge_buffer > 0) {
+      /* adjust bbox and width and height to the buffer */
+      buffer = map_edge_buffer * (map->extent.maxx - map->extent.minx) / (double)map->width;
+      map->extent.minx -= buffer;
+      map->extent.maxx += buffer;
+      map->extent.miny -= buffer;
+      map->extent.maxy += buffer;
+
+      map->width += 2 * map_edge_buffer;
+      map->height += 2 * map_edge_buffer;
+
+      /*
+      ** Ensure the labelcache buffer is greater than the buffer.
```

Comment:
I'm not sure about this one, and I haven't tested it myself as I'm not using any labels in this setup.
But it's copied from maptile.c so It shouldn't be too wrong.